base/collection.go

```diff
@@ -182,6 +183,7 @@ type Collection struct {
 	cluster          *gocb.Cluster // Associated cluster - required for N1QL operations
 	queryOps         chan struct{} // Manages max concurrent query ops
 	kvOps            chan struct{} // Manages max concurrent kv ops
+	collectionID     atomic.Value  // cached copy of collectionID
```

Comment:
I think this can just be *uint32, and use atomic.LoadUint32 and atomic.StoreUint32 for access.  The underlying implementation is faster, I think, and it also removes the type casting you're doing elsewhere.

Reply:
We decided to drop this - we don't want to use `sync.Once` since this operation might fail in the case of a network outage and we want to be able to retry getting the collection ID. A maybe more idiomatic API is to make this a struct with a mutex, but not going to pursue it for this code review.

The reason to use atomic.Value is that we need to check if `collectionID` is `nil` before you can call `atomic.LoadUint32` - and checking for nil obviates the use of atomic to prevent the race detection.